persistence/src/main/java/de/cyface/persistence/model/GeoLocationV6.java

```diff
@@ -0,0 +1,320 @@
+     */
+    @Ignore // Parcelable requires this constructor, make {@code Room} ignore this constructor.
+    protected GeoLocationV6(final @NonNull Parcel in) {
+        super(in);
+        lat = in.readDouble();
+        lon = in.readDouble();
+        altitude = in.readDouble();
+        speed = in.readDouble();
+        accuracy = in.readDouble();
+        verticalAccuracy = in.readDouble();
+        isValid = in.readByte() != 0;
+        measurementId = in.readLong(); // FIXME: Added, do we need this? GeoLocation has no such field
+    }
+
+    /**
+     * The {@code Parcelable} creator as required by the Android Parcelable specification.
+     */
+    public static final Creator<GeoLocationV6> CREATOR = new Creator<GeoLocationV6>() {
+        @Override
+        public GeoLocationV6 createFromParcel(Parcel in) {
+            return new GeoLocationV6(in);
+        }
+
+        @Override
+        public GeoLocationV6[] newArray(int size) {
+            return new GeoLocationV6[size];
+        }
+    };
+
+    @Override
+    public int describeContents() {
+        return 0;
+    }
+
+    @Override
+    public void writeToParcel(Parcel dest, int flags) {
+        super.writeToParcel(dest, flags);
+        dest.writeDouble(lat);
+        dest.writeDouble(lon);
+        dest.writeDouble(altitude);
+        dest.writeDouble(speed);
+        dest.writeDouble(accuracy);
+        dest.writeDouble(verticalAccuracy);
+        dest.writeByte((byte)(isValid ? 1 : 0));
+        dest.writeLong(measurementId); // FIXME: Added, do we need this? GeoLocation has no such field
+    }
+
+    @NonNull
+    @Override
+    public String toString() {
+        return "GeoLocationV6{" +
+                "lat=" + lat +
+                ", lon=" + lon +
+                ", altitude=" + altitude +
+                ", speed=" + speed +
+                ", accuracy=" + accuracy +
+                ", verticalAccuracy=" + verticalAccuracy +
+                ", isValid=" + isValid +
+                ", measurementId=" + measurementId +
+                '}';
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o)
+            return true;
+        if (o == null || getClass() != o.getClass())
+            return false;
+        GeoLocationV6 that = (GeoLocationV6)o;
+        return Double.compare(that.lat, lat) == 0
+                && Double.compare(that.lon, lon) == 0 && Double.compare(that.speed, speed) == 0
+                && Double.compare(that.accuracy, accuracy) == 0 && measurementId == that.measurementId
+                && Objects.equals(altitude, that.altitude) && Objects.equals(verticalAccuracy, that.verticalAccuracy)
+                && Objects.equals(isValid, that.isValid);
+    }
+
+    @Override
+    public int hashCode() { // FIXME: w/ or w/o uid, measurement_fk? (definitely w/o isValid)
+        return Objects.hash(lat, lon, altitude, speed, accuracy, verticalAccuracy);
+    }
```

Comment:
In `GeoLocation` we hashed:
- timestamp
- lat, lon, speed, accuracy
- isValid
The old class has no `measurementId` or database `identifier`, that data is only stored in the database, and in the case of `measurementId` only in the database and in `Measurement` and `Track` class, not `GeoLocation` directly.

Here in `GeoLocationV6 extends DataPointV6` we also have database `uid` and `measurementId`, but only after the data was loaded from the database, not while it's cached during capturing.

So the question would be: what to use for hashing?
- `DataPointV6.hash()` uses `uid` right now, but as `uid` is not set at cache-time I already questioned a few comments above if we should use `timestamp` for hashing, too, or maybe only `timestamp` as DataPoint(time=1L, mId=null) and `DataPoint(time=1L, mid=123L)` could considered the same object, just with and without mid set.

- `GeoLocationV6.hash()` should maybe use: `super.hash()` and `lat, lon, altitude, speed, accuracy, verticalAccuracy`, probably not `measurementId` as this is only set later but does not change the object "meant" by that object instance.

Reply:
I dimmly remember having put some thoughts into this years ago and not finding a satisfying architectural solution. Today I would probably consider to provide multiple classes or factor the changing parts out into their own class. But I think this is too much for this branch. As long as it works, a solution should only be added to the HEAD of main.